constant_fast_div.h

```diff
@@ -0,0 +1,106 @@
+/*
+* When dividing by a known compile time constant, the division can be replaced
+* by a multiply+shift operation. GCC will do this automatically, 
+* *BUT ONLY FOR DIVISION OF REGISTER-WIDTH OR NARROWER*.
+*
+* So on an 8-bit system, 16-bit divides will *NOT* be optimised.
+*
+* The macros here manually apply the multiply+shift operation for 16-bit numbers.
+*
+* Testing on an AtMega2560, -O3 optimizations:
+*   Performance improvement of 85% to 90%+ speed up (division by non-powers of 2)
+*   Zero increase in RAM usage
+*   Average of 95 bytes *less* Flash, than the equivalent GCC division code
+*  
+* Note: testing of the multiply+shift technique on 8-bit division showed a 
+* slight slow down over native code on AtMega2560. So the 8 bit equivalent 
+* macros have not been included
+*/
+
+#pragma once
+#include "libdivide.h"
+#include "u16_ldparams.h"
+#include "s16_ldparams.h"
+
+#define CAT_HELPER(a, b) a ## b
+#define CONCAT(A, B) CAT_HELPER(A, B)
+
+// GCC will optimise division by a power of 2
+// So allow that.
+#define S16_ISPOW2_NEG(denom) \
```

Comment:
any reason why 32768 is omitted in the below?

This might be clearer using shifts like `denom==(1<<6)` but up to you.